src/v3/src/util/settingsUtils.ts

```diff
@@ -107,3 +120,88 @@ export const getDefaultCountryCode = (widgetProps: WidgetProps): string => {
   return Object.keys(countries).includes(defaultCountryCode)
     ? defaultCountryCode : defaultCountry;
 };
+
+export const parseRegistrationSchema = (
+  widgetProps: WidgetProps,
+  schema: RegistrationElementSchema[],
+  onSuccess: RegistrationSchemaCallbackV3,
+  onFailure: RegistrationErrorCallback,
+): void => {
+  const { registration: { parseSchema } = {} } = widgetProps;
+  if (typeof parseSchema !== 'function') {
+    onSuccess(schema);
+    return;
+  }
+
+  parseSchema(
+    schema,
+    (modifiedSchema: RegistrationElementSchema[]) => onSuccess(modifiedSchema),
+    (error: APIError) => {
+      const errorObj = error || {
+        errorSummary: loc('registration.default.callbackhook.error', 'login'),
+      };
+      onFailure(errorObj);
+    },
+  );
+};
+
+export const preRegistrationSubmit = (
+  widgetProps: WidgetProps,
+  data: IdxActionParams,
+  onSuccess: RegistrationDataCallbackV3,
+  onFailure: RegistrationErrorCallback,
+): void => {
+  const { registration: { preSubmit } = {} } = widgetProps;
+  if (typeof preSubmit !== 'function') {
+    onSuccess(data);
+    return;
+  }
+
+  preSubmit(
+    data,
+    (postData) => onSuccess(postData),
+    (error: APIError) => {
+      const errorObj = error || {
+        errorSummary: loc('registration.default.callbackhook.error', 'login'),
+      };
+      onFailure(errorObj);
+    },
+  );
+};
+
+export const postRegistrationSubmit = (
+  widgetProps: WidgetProps,
+  response: string,
+  onSuccess: RegistrationPostSubmitCallback,
+  onFailure: RegistrationErrorCallback,
+): void => {
+  const { registration: { postSubmit } = {} } = widgetProps;
+  if (typeof postSubmit !== 'function') {
+    onSuccess(response);
+    return;
+  }
+
+  postSubmit(
+    response,
+    (responseStr) => onSuccess(responseStr),
+    (error: APIError) => {
+      const errorObj = error || {
+        errorSummary: loc('registration.default.callbackhook.error', 'login'),
+      };
+      onFailure(errorObj);
+    },
+  );
+};
+
+export const transformIdentifier = (
```

Comment:
Clarifying from yesterday's meeting, does this satisfy the `transformUsername` hook requirement?

Reply:
Yep! It does! 🙏 I updated the doc to reference this.